core/src/main/scala/com/pingcap/tispark/TiDBRelation.scala

```diff
@@ -83,28 +82,10 @@ case class TiDBRelation(session: TiSession,
     )
 
     val handlePlan = HandleRDDExec(tiHandleRDDs.toList)
-    // collect handles as a list
-    val aggFunc = CollectHandles(handlePlan.attributeRef.last)
-
-    val aggExpr = AggregateExpression(
-      aggFunc,
-      Complete,
-      isDistinct = true,
-      NamedExpression.newExprId
-    )
-
-    val sortAgg = TiUtil
-      .planAggregateWithoutPartial(
-        Seq(handlePlan.attributeRef.head), // group by region id
-        Seq(aggExpr),
-        Seq(handlePlan.output.head, aggExpr.resultAttribute), // output <region, handleList>
-        handlePlan
-      )
-      .children
-      .head
+    // TODO: we may optimize by partitioning the result by region.
```

Comment:
we need create an issue to track this. It is also better to have the issue link.